components/compositing/compositor.rs

```diff
@@ -1028,16 +1055,21 @@ impl<Window: WindowMethods + ?Sized> IOCompositor<Window> {
         }
     }
 
-    /// Set the root pipeline for our WebRender scene. If there is no pinch zoom applied,
-    /// the root pipeline is the root content pipeline. If there is pinch zoom, the root
-    /// content pipeline is wrapped in a display list that applies a pinch zoom
-    /// transformation to it.
-    fn set_root_content_pipeline_handling_device_scaling(&self, transaction: &mut Transaction) {
-        let root_content_pipeline = match self.root_content_pipeline.id {
-            Some(id) => id.to_webrender(),
-            None => return,
-        };
+    /// Set the root pipeline for our WebRender scene to a display list that consists of an iframe
+    /// for each visible top-level browsing context, applying a transformation on the root for
+    /// pinch zoom, page zoom, and HiDPI scaling.
+    fn update_root_pipeline(&mut self) {
+        let mut transaction = Transaction::new();
+        self.update_root_pipeline_in_transaction(&mut transaction);
+        self.generate_frame(&mut transaction, RenderReasons::SCENE);
+        self.webrender_api
+            .send_transaction(self.webrender_document, transaction);
+    }
 
+    /// Set the root pipeline for our WebRender scene to a display list that consists of an iframe
+    /// for each visible top-level browsing context, applying a transformation on the root for
+    /// pinch zoom, page zoom, and HiDPI scaling.
+    fn update_root_pipeline_in_transaction(&self, transaction: &mut Transaction) {
```

Comment:
I think this could be a bit more descriptive to make it obvious what this method is doing. How do you feel about:

```suggestion
    fn send_root_pipeline_display_list_in_transaction(&self, transaction: &mut Transaction) {
```

Reply:
done in b2528dc1bbe43ea18822c51cc3c2d6c10222bc0e